pkg/charts/utils.go

```diff
@@ -166,5 +173,12 @@ func generateChartValues(config *ciliumv1alpha1.NetworkConfig, network *extensio
 	if config.BPFSocketLBHostnsOnly != nil {
 		globalConfig.BPFSocketLBHostnsOnly.Enabled = config.BPFSocketLBHostnsOnly.Enabled
 	}
+
+	// If node local dns feature is enabled, enable local redirect policy
+	if cluster.Shoot.Annotations[v1beta1constants.AnnotationNodeLocalDNS] == "true" {
+		globalConfig.NodeLocalDNS.Enabled = true
+		globalConfig.LocalRedirectPolicy.Enabled = true
+	}
+
```

Comment:
Please move this block before the `if config == nil {` check in line 145. Otherwise, it will only be executed if a special config is set, which is not the expectation.